test/functional/c/test_delaunay.cpp

```diff
@@ -21,10 +24,60 @@ TEST(Delaunay, Delaunay_reduce_layer) {
     del_layer_delaunay_reduce(min_lattice, lattice, aperiodic_axis, symprec);
 
     // Shortest vectors in the periodic plane are [0, 1, 0] and [0, 0, 1]
-    debug_print_matrix_d3(min_lattice);
     for (int i = 0; i < 3; ++i) {
         double sum = fabs(min_lattice[0][i]) + fabs(min_lattice[1][i]) +
                      fabs(min_lattice[2][i]);
-        ASSERT_FLOAT_EQ(sum, 1);
+        EXPECT_FLOAT_EQ(sum, 1);
+    }
+}
+
+TEST(Delaunay, Delaunay_reduce) {
+    double min_lattice[3][3];
+    double lattice[3][3] = {
+        {-2.2204639179669590, -4.4409278359339179, 179.8575773553236843},
+        {1.2819854407640749, 0.0, 103.8408207018900669},
+        {10.5158083946732219, 0.0, 883.3279051525505565},
+    };
+    double inv_lat[3][3], tmat[3][3], metric[3][3];
+    int int_tmat[3][3];
+
+    auto setenv_result = setenv("SPGLIB_NUM_ATTEMPTS", "100", 1);
+    ASSERT_EQ(setenv_result, 0);
+
+    const double symprec = 1e-5;
+    int succeeded = del_delaunay_reduce(min_lattice, lattice, symprec);
+    // Default get_num_attempts() == 1000 --> succeeded == 1.
+    // With get_num_attempts() == 100 --> succeded == 0.
+    EXPECT_EQ(succeeded, 0);
+    auto error = spg_get_error_code();
+    EXPECT_EQ(error, SPGERR_DELAUNAY_FAILED);
```

Comment:
@atztogo @lan496, actually this one fails, and instead `SPGLIB_SUCCESS` is reported, which is too unexpected. How about adding more of the relevant `spg_set_error_code` in the inner functions?

Reply:
The implementation would not be hard. The question is, what error code should these output? `SPGERR_DELAUNAY_FAILED`, or a new one `SPGERR_MAX_ATTEMPTS`/`SPGERR_DELAUNAY_MAX_ATTEMPTS`?

Reply:
I have given this more thought. These functions are not exposed publicly, so there is no need to check for errors at that stage. Ideally if we can migrate to C++/rust we can trigger and propagate the errors from inside out, but for now I will revert this check.

Reply:
Thanks @LecrisUT for your comments. Can this PR be merged?

Reply:
:+1: Yes